### PR TITLE
ServiceControl Integration added additional events

### PR DESCRIPTION
**Pull request by seanfarmar**
Related to https://github.com/Particular/ServiceControl/pull/2465

### CODE REVIEW

**Review by danielmarbach — APPROVED** [review]
Will you also update the sample? As a different PR?

**Comment by seanfarmar** [comment]
> Will you also update the sample? As a different PR?

I have an updated version of the sample, but for it to work I will need to update the platform sample with the serviceControl.Contracts...

Will I do this after the release?

File: samples/servicecontrol/azure-monitor-events/sample.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+Running the project will result in 3 console windows:
+
+1. **NServiceBusEndpoint**: The endpoint that represents the system being monitored.
+1. **AzureMonitorConnector**: The endpoint that subscribes to ServiceControl notification events and pushes them to Application Insights as custom telemetry events.
+1. **PlatformLauncher**: Runs an in-process version of ServiceControl and ServicePulse. When the ServiceControl instance is ready, a browser window will be launched displaying the ServicePulse Dashboard.
+
+The samples enables triggering two types of events:
+
+### MessageFailed event
```
Comment (seanfarmar) [inline samples/servicecontrol/azure-monitor-events/sample.md:30]:
```suggestion
### Message Failures
```

File: samples/servicecontrol/azure-monitor-events/sample.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+When a `MessageFailed` event is received, the `AzureMonitorConnector` prints the following message in its console window: 
+
+```
+> Received ServiceControl 'MessageFailed' event for a SimpleMessage with ID 42f25e40-a673-61f3-a505-c8dee6d16f8a
+```
+
+The failed message can also be viewed in the ServicePulse browser window. Navigating to the failed message allows viewing more details about the message failure.
+
+
+### HeartbeatStopped and HeartbeatRestored events
```
Comment (seanfarmar) [inline samples/servicecontrol/azure-monitor-events/sample.md:47]:
```suggestion
### Heartbeat statuses
```

File: samples/servicecontrol/azure-monitor-events/sample.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+component: ServiceControlContracts
+reviewed: 2021-04-16
+related:
+ - servicecontrol
+ - servicecontrol/contracts
+ - servicecontrol/plugins
+ - samples/servicecontrol/monitoring3rdparty
+---
+
+This sample shows how to monitor activities performed in ServiceControl and ServicePulse using the integration events. The sample uses the [Learning Transport](/transports/learning/) and a portable version of the Particular Service Platform tools. Installing ServiceControl is **not** required.
```
Comment (seanfarmar) [inline samples/servicecontrol/azure-monitor-events/sample.md:13]:
```suggestion
This sample shows how to monitor a running NServiceBus system with ServiceControl and ServicePulse, as well as how to integrate with existing monitoring solutions. The sample uses the [Learning Transport](/transports/learning/) and a portable version of the Particular Service Platform tools. Installing ServiceControl is **not** required.
```

**Review by danielmarbach — APPROVED** [review]
I think what is missing is the documentation of the flag to disable the event publishing on the settings/configuration page (can also be done in another PR). Other than that this looks good to me

File: samples/servicecontrol/azure-monitor-events/ServiceControlContracts_3/EndpointsMonitor/CustomEventsHandler.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using NServiceBus;
+using NServiceBus.Logging;
+using ServiceControl.Contracts;
+
+#region AzureMonitorConnectorEventsHandler
```
[the remark this answers is not in the document]
Comment (tmasternak) [reply samples/servicecontrol/azure-monitor-events/ServiceControlContracts_3/EndpointsMonitor/CustomEventsHandler.cs:10]: We wanted to show the whole handler including the telemetry client injection and the message name. To make it smaller we moved all other supported messaged to a `CustomEventsHandler`.

**Review by tmasternak — APPROVED** [review]

**Pull request merged**